File: AlgorithmsInPython/LinearSearch.py
# Linear Search in Python



def linearSearch(array, n, key):
    """
    Linear Search Algorithm

    LinearSearch(array, key)
    for each item in the array
        if item == value
        return its index

    Time Complexity: O(n)

    Space Complexity: O(1)

    """

    for i in range(0, n):
        if (array[i] == key):
            return i
    return -1


array = [7, 9, 11, 55, 13, 3, 0, 49]
x = 13
n = len(array)
result = linearSearch(array, n, x)
if(result == -1):
    print("Element not found")
else:
    print("Element is: ", array[result])
    print("Element found at index: ", result)

File: AlgorithmsInC/LinearSearch.c
// Linear Search

/*

Time Complexity: O(n)
Space Complexity: O(1)

LinearSearch(array, key)
  for each item in the array
    if item == value
      return its index
*/

#include <stdio.h>
#include <stdlib.h>

int linearSearch(int array[], int n, int key)
{
  for (int i = 0; i < n; i++)
    if (array[i] == key)
      return i;
  return -1;
}

int main()
{
  int array[] = {2, 4, 60, 8, 3, 5, 1};
  int x = 3;
  int n = sizeof(array) / sizeof(array[0]);

  int result = search(array, n, x);

  (result == -1) ? printf("Element not found.") : printf("Element found at index: %d\nElement is = %d", result, array[result]);
}
